log/sequencer.go

```diff
@@ -112,22 +113,32 @@ func (s Sequencer) SequenceBatch(limit int) error {
 	if err != nil {
 		glog.Warningf("Sequencer failed to get latest root: %s", err)
 		tx.Rollback()
-		return err
+		return 0, err
 	}
 
-	// Initialize the compact tree state to match the latest root in the database
-	mt, err := s.buildMerkleTreeFromStorageAtRoot(currentRoot, tx)
+	var mt *merkle.CompactMerkleTree
+
+	if currentRoot.TreeSize == 0 {
+		// This should be an empty tree
+		if currentRoot.TreeRevision > 0 {
+			panic(fmt.Errorf("MT has zero size but non zero revision: %v", *mt))
```

Comment:
Yeah for the moment I wanted to halt the server because this is a "can't happen" case. I'll todo it to downgrade.